### ISSUE TITLE
Does not handle non-unique names

### ISSUE THREAD

**Issue by auscompgeek** [wee-slack/wee-slack#685]
wee-slack currently does not make conflicting names unique on the client-side, making it impossible to:
- correctly determine who is in a channel,
- ping the users with non-unique names,
- ...

**Comment by trygveaa** [comment]
Do you have any suggestions for how to best solve this? Not technically, but from a UI point of view. I guess we could include the real name in the nick or something, but that might make for long nicks.

**Comment by auscompgeek** [comment]
I'd be happy with something like simply trying appending an underscore or an incrementing number until there's no conflict.

**Comment by trygveaa** [comment]
A potential issue with just appending an underscore or a number is that it may be confusing when people join or leave.

If we just assign underscore or a number randomly, it may change when wee-slack is reloaded, so we probably want to assign them in order based on e.g. alphabetical order of full name or email address. However, if you have a channel with user1 and user2, and then a third person with the same name joins which would be ordered first, then we have to rename user1 to user2 and user2 to user3, which would probably be confusing for the user.

**Comment by apetresc** [comment]
> However, if you have a channel with user1 and user2, and then a third person with the same name joins which would be ordered first, then we have to rename user1 to user2 and user2 to user3, which would probably be confusing for the user.

Well, couldn't we do the suffix globally for the entire Team? Then, yes, you would still have to do a reshuffle whenever a new person with the exact same display name joins the team, but surely that's an extremely rare event. But let's not let perfect be the enemy of the good - the current state is confusing and potentially dangerous[^1] _right now_ in every single case, instead of only sometimes when an anomalous event occurs.

[^1]: Example of what I mean by "dangerous": I'm in a channel with someone named `@Matt`. I needed to send him a secret credential out-of-channel so I just `/msg Matt` it to him. Every IRC-based instinct I've developed over the last 20 years told me this should be a fine and safe thing to do. It ended up going to a completely different `@Matt` on the team (a remote sales guy who, to be honest, I didn't even remember existed).

**Comment by trygveaa** [comment]
> Well, couldn't we do the suffix globally for the entire Team?

Yeah, it would be for the whole team, so it might not be a big issue.

> But let's not let perfect be the enemy of the good

Sure, I just wanted to discuss the potential issues and hear if anyone had any better solutions. I agree that appending underscores or numbers is better than the current behavior.
